File: resources/exercises/sort/basic/insertion/AlgInsertionSortEntity.py
# BEGIN SOLUTION
for i in range(1,getValueCount()):
  value = getValue(i)
  j = i
  while j>0 and not isSmallerThan(j-1,value):
    copy(j-1,j)
    j = j-1
  setValue(j,value)
# END SOLUTION

File: resources/exercises/sort/basic/insertion/AlgInsertionSortEntity.c
//RemoteSort

/* BEGIN TEMPLATE */
void insertionSort() {
	/* BEGIN SOLUTION */
	int i;
	for (i = 1; i < getValueCount(); i++) {
		int value = getValue(i);
		int j = i;
		while ((j > 0) && (!isSmallerThan(j-1,value))) {
			copy(j-1,j);
			j--;
		}
		setValue(j,value);
	}
	/* END SOLUTION */
}
/* END TEMPLATE */



void run(){
	insertionSort();
}
